### PR TITLE
Implemented control interface to issue commands to running instance

### LINKED ISSUE #196: Unix socket based, command listener implementation
For the purposes of convenient management of dynamic settings of the application, there's a proposal to implement unix socket listener, which will listen for external commands, and to dynamically change the settings accordingly. It will be quite flexible alternative for signal handling.

### PR DESCRIPTION
**Pull request by bobs4462**
- Moved CLI options to a separate module
- Added extra dependency: awc-uds, in order to perform http requests
  over unix sockets, as awc doesn't seem to support it out of box
- 2 Commands are supported for now: config-reload, subscriptions on/off

Closes #196

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -44,6 +44,7 @@ humantime = "2.1"
 toml = "0.5"
 arc-swap = "1.4"
 mlua = { version = "0.6", features = ["luajit", "vendored"] }
+awc-uds = { version = "0.1.0", git = "https://github.com/polachok/awc-uds.git", rev = "9dafa17"}
```
Comment (polachok) [inline Cargo.toml:47]: 29c9bd24791

File: src/cli_options.rs [hunk 2]
```diff
@@ -0,0 +1,189 @@
+use anyhow::Result;
```
Comment (polachok) [inline src/cli_options.rs:1]: I suggest renaming this cli_options -> cli

File: src/control_interface.rs [hunk 3]
```diff
@@ -0,0 +1,157 @@
+use std::{
```
Comment (polachok) [inline src/control_interface.rs:1]: control_interface -> control

File: src/control_interface.rs [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    .unwrap();
+}
+
+#[derive(Deserialize, Debug)]
+struct SubscribeAction {
+    action: String,
+}
+
+#[post("/subscriptions/{action}")]
+async fn subscriptions_setter(
```
Comment (polachok) [inline src/control_interface.rs:78]: Would be nice to also have getter

File: src/control_interface.rs [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        let config = std::fs::File::open(&sender.path)
+            .map(Config::from_file)
+            .map_err(|_| ControlError::BadConfigFile)?
+            .map_err(|_| ControlError::BadConfigFile)?;
+
+        sender
+            .tx
+            .send(config.rpc)
+            .map_err(|_| ControlError::NoConfigFile)?;
+        Ok(HttpResponse::Ok().body(AnyBody::Bytes(Bytes::copy_from_slice(b"OK"))))
```
Comment (polachok) [inline src/control_interface.rs:112]: AnyBody::from_slice ?

File: src/main.rs [hunk 6]
```diff
@@ -176,63 +74,6 @@ async fn main() -> Result<()> {
...
-
-async fn listen_for_subscription_limit(subscriptions_allowed: Arc<AtomicBool>) {
-    let mut stream =
-        signal(SignalKind::user_defined1()).expect("failed to register signal handler for USR1");
-    while stream.recv().await.is_some() {
-        subscriptions_allowed.fetch_xor(true, Ordering::Relaxed);
-    }
-}
-
-async fn config_read_loop(path: PathBuf, rpc: watch::Sender<rpc::Config>) {
```
Comment (polachok) [inline src/main.rs:217]: This one shouldn't be removed for compatibility

**Review by polachok — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 6 above)
Comment (bobs4462) [reply src/main.rs:217]: Reverted it back

File: src/control_interface.rs (hunk 5 above)
Comment (bobs4462) [reply src/control_interface.rs]: :heavy_check_mark: Done

File: src/control_interface.rs (hunk 4 above)
Comment (bobs4462) [reply src/control_interface.rs]: :heavy_check_mark: Done

File: src/control_interface.rs [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+use tokio::sync::watch::Sender;
+use tracing::{info, warn};
+
+use crate::{
+    cli_options::{Command, Config},
+    rpc,
+};
+
+pub const CACHER_SOCKET: &str = "/tmp/cacher.sock";
+
```
[the remark this answers is not in the document]
Comment (bobs4462) [reply src/control_interface.rs]: :heavy_check_mark: Done

**Review by polachok — APPROVED** [review]

**Pull request merged**